pallets/messages/src/lib.rs

```diff
@@ -180,20 +184,25 @@ pub mod pallet {
 		/// * `payload_length` - The size of the payload
 		/// * Returns
 		/// * [DispatchResultWithPostInfo](https://paritytech.github.io/substrate/master/frame_support/dispatch/type.DispatchResultWithPostInfo.html)
-		#[pallet::weight(T::WeightInfo::add_ipfs_message(payload.cid.get().len() as u32, 1_000))]
+		#[pallet::weight(T::WeightInfo::add_ipfs_message(cid.len() as u32, 1_000))]
 		pub fn add_ipfs_message(
 			origin: OriginFor<T>,
 			on_behalf_of: Option<MessageSourceId>,
 			schema_id: SchemaId,
-			payload: IPFSPayload,
+			cid: Vec<u8>,
+			payload_length: u32,
```

Comment:
As per @enddynayn 's suggestion, I don't think we need a dedicated struct for the payload (at least not yet).